src/Moq/Mock.Generic.cs

```diff
@@ -147,6 +147,21 @@ public Mock(MockBehavior behavior, params object[] args)
 			this.CheckParameters();
 		}
 
+		/// <summary>
+		///   Initializes an instance of the mock using the given constructor call including its
+		///   argument values and with a specific <see cref="MockBehavior"/> behavior.
+		/// </summary>
+		/// <param name="newExpression">Lambda expression that creates an instance of <typeparamref name="T"/> T.</param>
+		/// <param name="behavior">Behavior of the mock.</param>
+		/// <example>
+		/// <code>var mock = new Mock&lt;MyProvider&gt;(() => new MyProvider(someArgument, 25));</code>
+		/// </example>
+		[SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors")]
+		public Mock(Expression<Func<T>> newExpression, MockBehavior behavior = MockBehavior.Strict)
```

Comment:
* `behavior` must default to `MockBehavior.Default`. It'll just be confusing if Moq sometimes creates strict mocks by default, but loose mocks in other places.

* Moq's public API in general does not use optional parameters, but instead sets default values through different overloads and constructor / method chaining. It would be great if you could follow that pattern (for consistency).

Reply:
Tried your second suggestion - unfortunately this creates once again a ambiguity with Mock<T>.Mock(params object[])...